### DISCUSSION TITLE
last page

### DISCUSSION THREAD

**Discussion by smj1234** [pocketbase/pocketbase#5309] [Q&A] [answered]
sry to create this - i searched all the docs and i couldn't find solution to this matter 

i have countless Datas in pocket base , in each collection i have lost of row (+500.000 )

unfortunately, when i want to get first page it gives me random data and when i use sorting for ex -Date or -number takes a lot of time to sort (about 5 sec) i checked with higher cores , ram and even higher generations, the result was same (i should say that the resource usage is less than 1%)

but when i sort with Date or Number (without "-" in the back of it ) it take about 300ms or less to sort and give 1st page 

so the problem starts here - i cant get the last page str8
i need to req 2 time once for the all pages (my data counted from old to new so the numbers go up when i add new Datas (for ex number 100 is newer than but num 0 that why i need to sort from high to low in numbers) 

wonder is there a way to do this with 1 req ? i rly need to reduce the amount of req I'm doing cause the Datas are refreshing few times per minute

**Comment by ganigeorgiev** [comment]
Without details about your collections structure and the specific request it is difficult to reproduce or even understand your problem.

If you are experiencing issues with the performance then you can try adding individual indexes for the specific field you are sorting by (under the "Unique constraints and indexes" section in the collection "Fields" tab).

Alternatively, you can also start the executable with `--dev`, find the slow query and try to translate it into a plain SQL statement. Then you can execute it with the `sqlite3 pb_data/data.db` to check what route the query planner has chosen and whether your index has any effect (aka. `EXPLAIN QUERY PLAN yourQuery`).

If you need further help, then please try to elaborate more clearly what actually is the issue, provide an export of your collections structure (_Admin UI > Settings > Export collections_) and a code sample with the problematic request.

**Reply by smj1234** [reply]
hi rly appreciate you help 

my collection is simple i don't know how  to add  it here so i just uploading the Json

[pb_schema (2).json](https://github.com/user-attachments/files/16479257/pb_schema.2.json)


the code is simple too i just using post man for testing 

first pic shows the sorting with candle_number
![ScreenShot_20240803143646](https://github.com/user-attachments/assets/4012be58-fffa-44a2-bf22-14496f1f3813)

sec pic shows the sorting with -candle_number
![ScreenShot_20240803143723](https://github.com/user-attachments/assets/fc1328dd-16c3-458d-bc31-a73869509c2e)

and you can see the performance here 

wonder if there is a way to get last page str8 or if its not 
how you will suggest to do this database to get beter performance sir? i also checked the Json field to use it but no luck ... !


also i apologies for bad Eng

**Reply by ganigeorgiev** [reply, marked as answer]
> wonder if there is a way to get last page str8 or if its not
how you will suggest to do this database to get beter performance sir? i also checked the Json field to use it but no luck ... !

Honestly I don't understand what do you mean by _"get last page str8"_ and what the `json` field type has to do with anything (shortening words and phrases and the lack of capital letters and punctuation doesn't help either).

The above schema export is also incomplete (it seems that it is missing the "Pair" collection).

In any case to improve the performance you can try to:

- Add index on the `Candle_Number` field:
  ![index_screenshot](https://i.imgur.com/dhz1Qdm.png)

- Add the `skipTotal=1` query parameter. This will skip the extra `COUNT` query that is executed by default with the list request to generate the `totalItems` and `totalPages` values.

More than this I don't think I can help further.

**Reply by smj1234** [reply]
i should say wow I'm surprised it works !!!! 


![ScreenShot_20240803150448](https://github.com/user-attachments/assets/201a8381-c391-444d-bc85-0c2ca00dfef8)


by json field i meant to reduce the amount of data and compact them into json 

the schema  was only one collection it was empty ether so i requested on other collection for postman pic


i think its better to add this tips on doc somewhere

thx for help i rly in your debt  :X
